src/components/CommentForm.vue

```
@@ -0,0 +1,140 @@
+          placeholder="Write your comment..."
+        ></textarea>
+      </div>
+      <p v-if="submitted && !body" class="help is-danger">Comment text is required</p>
+    </div>
+
+    <div class="field is-grouped">
+      <div class="control">
+        <button
+          type="submit"
+          class="button is-link"
+          :class="{ 'is-loading': loading }"
+        >
+          Submit
+        </button>
+      </div>
+      <div class="control">
+        <button
+          type="button"
+          class="button is-light"
+          @click="clearForm"
+          :disabled="loading"
+        >
+          Clear
+        </button>
+      </div>
+    </div>
+  </form>
+</template>
+
+<script>
+export default {
+  name: 'CommentForm',
+  props: {
+    postId: {
+      type: Number,
+      required: true,
+    },
+    onSubmit: {
+      type: Function,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      name: '',
+      email: '',
+      body: '',
+      loading: false,
+      submitted: false,
+    };
+  },
+  methods: {
+    clearForm() {
+      this.name = '';
+      this.email = '';
+      this.body = '';
+      this.submitted = false;
+    },
+
+    async handleSubmit() {
+      this.submitted = true;
+
+      if (!this.name || !this.email || !this.body) {
+        return;
+      }
+
+      this.loading = true;
+
+      try {
+        const newComment = await this.onSubmit({
+          name: this.name,
+          email: this.email,
+          body: this.body,
+        });
+
+        // mantém nome e e-mail, limpa apenas o texto
+        this.body = '';
+        this.submitted = false;
+
+        // opcional: foco no campo de comentário novamente
+        this.$nextTick(() => {
+          const textarea = this.$el.querySelector('textarea');
+          if (textarea) textarea.focus();
+        });
+
+        // append é feito no App.vue ou componente pai
+      } catch (error) {
+        console.error('Failed to submit comment', error);
```

The catch block only logs the error to console. According to the task you must "Handle Add and Delete errors (for comments) so the user can retry". Swallowing the error prevents the parent from showing an error/allowing retry. Consider one of the following:

- Re-throw the error after logging so a parent `await`-ing the call can catch it and set a comments error notification, or
- Emit an event like `this.$emit('failed', error)` or `this.$emit('error', 'Failed to add comment')` so the parent can present a retry UI, or
- Set a local error message and render a Notification inside the form.

Also make sure to keep `loading` usage consistent so the submit button has `is-loading` while waiting.